src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java

```diff
@@ -96,6 +96,7 @@ public AuthorizationMatrixProperty(Map<Permission,? extends Set<String>> granted
             this.grantedPermissions.put(e.getKey(),new HashSet<>(e.getValue()));
     }
 
+    @Restricted(NoExternalUse.class)
```

Comment:
Why? I understand that nobody is expected to write to this method. `Collections.unmodifyableSet()` would be reasonable, but I do not see issues with having getter in API

Reply:
This only exists to build up the potential groups for http://javadoc.jenkins.io/hudson/security/AuthorizationStrategy.html#getGroups-- and is not intended for any other use.

Reply:
Collections.unmodifyableSet() would be useful anyway

Reply:
I went with a copy of the set instead, consistent with some other getters that return a newly built collection.